charts/ckan/templates/ckan/nginx-configmap.yaml

```diff
@@ -0,0 +1,137 @@
+{{- if .Values.ckan.nginx.configMap.create }}
+{{- $fullName := (print .Release.Name "-ckan") -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ $fullName }}-nginx-conf
+  labels:
+    app: {{ $fullName }}
+    app.kubernetes.io/name: {{ $fullName }}
+    app.kubernetes.io/component: nginx
+data:
+  nginx.conf: |-
+
+    error_log  /dev/stderr warn;
+    pid        /tmp/nginx.pid;
+
+    events {
+      worker_connections  1024;
+    }
+
+    http {
+      include       /etc/nginx/mime.types;
+      default_type  application/octet-stream;
+
+      client_body_temp_path /tmp/client_temp;
+      proxy_temp_path       /tmp/proxy_temp_path;
+      fastcgi_temp_path     /tmp/fastcgi_temp;
+      uwsgi_temp_path       /tmp/uwsgi_temp;
+      scgi_temp_path        /tmp/scgi_temp;
+
+      proxy_buffer_size 16k;  # Max total size of response headers.
+      # n * m = max response size before spooling to disk. p95 response size should
+      # fit comfortably within this in order to avoid performance issues.
+      proxy_buffers 24 16k;
+
+      server_tokens off;
+
+      sendfile        on;
+      keepalive_timeout  65;
+
+      # Set GOVUK-Request-Id if not set
+      map $http_govuk_request_id $govuk_request_id {
+        default $http_govuk_request_id;
+        ''      "$pid-$msec-$remote_addr-$request_length";
+      }
```

Comment:
You can drop govuk-request-id (here and in the log format and proxy-set-header below) — CKAN doesn't use it. (It's only useful for GOV.UK's own apps where it gets forwarded to backends so you can do a rudimentary kind of request tracing.)